DasharoPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c

```diff
@@ -1567,6 +1582,19 @@ PlatformBootManagerAfterConsole (
   EfiBootManagerConnectAll ();
   EfiBootManagerRefreshAllBootOption ();
 
+  //
+  // Process device update capsules there weren't processed along with system
+  // firmware capsules on first call to ProcessCapsules() in
+  // PlatformBootManagerBeforeConsole().
+  //
+  if (GetBootModeHob() == BOOT_ON_FLASH_UPDATE) {
+    // TODO: when enabling capsule support for laptops, add a battery check here
+    Status = ProcessCapsules ();
+    if (EFI_ERROR (Status)) {
+      DEBUG((DEBUG_ERROR, "%a(): ProcessCapsule() failed with: %r\n", __FUNCTION__, Status));
+    }

```

Comment:
We have a requirement for not allowing to boot to OS with SMI handler enabled, this seems to be a good place to enforce it.

```suggestion
    }
    // Reset system to disable SMI handler
    gRT->ResetSystem (EfiResetCold, EFI_SUCCESS, 0, NULL);
```

Reply:
https://github.com/Dasharo/edk2/compare/e4bb5bf01ba0efbe91fe702bfb08dd9d47b9f174..4fd97ed77cc98d27a2465d7b5898daa8908f0028